lib/cores.js

```diff
@@ -108,36 +109,45 @@ async function getConnectedCoreIPsAsync(coreIPList, coreConnectionCount) {
   }
 }
 
-async function coreRequestAsync(options, coreIP, retryCount = 3) {
+async function coreRequestAsync(options, coreIP, retryCount = 3, timeout = 500) {
   options.headers['X-Node-Version'] = version
   options.uri = `http://${coreIP}${options.uriPath}`
 
   let response
   if (retryCount <= 0) {
-    response = await rp(options)
+    try {
+      response = await rp(options)
+    } catch (error) {
+      if (error.statusCode === 402 || error.status === 402) return error.response
```

Comment:
All error codes throw with request-promise, so we need to catch it and return with headers for 402